### ISSUE TITLE
Trying to make a simple select never returns data

### ISSUE THREAD

**Issue by bwship** [supabase-community/supabase-swift#95]
# Bug report

## Describe the bug

I have a simple piece of code.  My auth is working, and upserts work fine as well.  But no matter what I do, trying to select data fails.

## To Reproduce

            var profile = Basic(id: 2)
            try await client.database
                            .from("test")
                            .upsert(values: profile)
                            .execute()

             var data = try await client.database.from("test")
                                        .select()
                                        .execute().value
            
            print(data)

This is the snippet of code I am using.  The upsert works, but data is never returned.

## Expected behavior

This should returns data from the table.

## System information

- OS: MacOS
- Version of supabase-swift: 0.3.0

**Comment by bwship** [comment]
For what it's worth, it is returning a 200.  And if I do a curl command with the url it is hitting, that returns back 2 records.

**Comment by grsouza** [comment]
Hey @bwship thanks for opening this issue, there are a couple of things that may be happening here.

1. Can you double-check If your RLS policies are correctly defined allowing to query data? As you can get data using the cURL I suppose they are correct, but double-check if you're using the `anonymous key` on both cases.
2. Are you casting the result of `value` property to your desired type? For example:

```diff
+  var data: [Model] = try await client.database.from("test")
-  var data = try await client.database.from("test")
                                    .select()
                                    .execute().value
```

**Comment by bwship** [comment]
I definetely have the RLS policies turned off.

When I try to do this

var data: [Basic] = try await client.database.from("test")
                .select().execute().value

I get a compile time error of "Cannot convert value of type 'Void' to specified type [Basic].  I changed it to 

struct Basic: Decodable, Encodable {
    var id: Int
}

var data: [Basic]? = try await client.database.from("test")
                .select().execute().value

Adding the optional ? at the end and then making the Basic struct be Decodable, and BOOM, it worked!  Thanks, that is wild I had to do the type in there, as it seemed like even if I just did the .execute() and then looked at the .count value, it would always show nil.  But thanks so much for the help!
